File: src/main/java/com/kelsos/mbrc/events/ui/DisplayDialog.java
package com.kelsos.mbrc.events.ui;

public class DisplayDialog {
    public static final int NONE = 0;
    public static final int SETUP = 1;
    public static final int UPGRADE = 2;
    public static final int INSTALL = 3;

    private int dialogType;

    public DisplayDialog(int dialogType) {
        this.dialogType = dialogType;
    }

    public int getDialogType() {
        return dialogType;
    }
}

File: app/src/main/kotlin/com/kelsos/mbrc/events/ui/DisplayDialog.kt
package com.kelsos.mbrc.events.ui

import android.support.annotation.IntDef
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

class DisplayDialog(@DialogType val dialogType: Int) {

  @IntDef(NONE.toLong(), SETUP.toLong(), UPGRADE.toLong(), INSTALL.toLong())
  @Retention(RetentionPolicy.SOURCE)
  annotation class DialogType

  companion object {
    val NONE = 0
    val SETUP = 1
    val UPGRADE = 2
    val INSTALL = 3
  }
}
